R/param-estimates-batch.R

```diff
@@ -110,51 +110,117 @@ param_estimates_batch <- function(.path,
-#'   estimates in `.param_df`.
+#'   estimates in `.boot_sum`.
 #' @param na.rm Logical scalar, passed through to [stats::quantile()].
 #'
 #' @importFrom tibble rownames_to_column
 #' @importFrom dplyr summarise across starts_with select rename left_join
 #' @importFrom stats quantile
+#'
+#' @return A tibble containing quantiles for each parameter estimate, optionally
+#' compared to the estimates from `.orig_mod`.
+#' @seealso summarize_bootstrap_run
+#' @examples
+#' \dontrun{
+#'
+#' # Via a bootstrap run
+#' boot_run <- read_model(file.path(MODEL_DIR, "1-boot"))
+#' boot_sum <- summarize_bootstrap_run(boot_run)
+#' param_estimates_compare(boot_sum)
+#'
+#' # Via a custom table
+#' orig_mod <- read_model(file.path(MODEL_DIR, "1"))
+#' param_df <- param_estimates_batch(MODEL_DIR)
+#' param_estimates_compare(param_df, .orig_mod = orig_mod)
+#' }
+#'
 #' @export
 param_estimates_compare <- function(
-  .param_df,
-  .orig_mod = NULL,
-  .compare_cols = starts_with(c("THETA", "SIGMA", "OMEGA")),
-  probs = c(.5, 0.025, 0.975),
-  na.rm = FALSE
-) {
-
-  comp_df <- .param_df %>%
+    .boot_sum,
+    .orig_mod = NULL,
+    .compare_cols = starts_with(c("THETA", "SIGMA", "OMEGA")),
+    probs = c(.5, 0.025, 0.975),
+    na.rm = FALSE
+){
+  UseMethod("param_estimates_compare")
+}
+
+#' @rdname param_estimates_compare
+#' @export
+param_estimates_compare.bbi_nmboot_summary <- function(
+    .boot_sum,
+    .orig_mod = NULL,
+    .compare_cols = NULL,
+    probs = c(.5, 0.025, 0.975),
+    na.rm = FALSE
+){
+
+  # Attempt to read in based_on model
+  if(is.null(.orig_mod)){
+    orig_mod_path <- fs::path_ext_remove(.boot_sum$based_on_model_path)
+    # make sure based_on model still exists
+    .orig_mod <- tryCatch(
+      read_model(orig_mod_path),
+      error = function(cond) NULL
+    )
+    if(!is.null(.orig_mod)){
+      orig_config_path <- file.path(
+        get_output_dir(.orig_mod, .check_exists = FALSE), "bbi_config.json"
+      )
+      if (fs::file_exists(orig_config_path)) {
+        res <- check_up_to_date(.orig_mod)
+        if(any(res[!res])){
+          rlang::warn("The original model is not up to date")
+        }
+      }else{
+        # Set the original model to NULL if the run directory/config file didnt
+        # exist
+        rlang::warn("The original model has not been run yet. Cannot compare estiamtes.")
+        .orig_mod <- NULL
+      }
+    }else{
+      rlang::warn(
+        c(
+          glue("The original model no longer exists at {orig_mod_path}"),
+          "Cannot compare to original model"
+        )
+      )
+    }
+  }
+
+  # Dont pass .compare_cols here, as we can only use columns in parameter_names,
+  # which could only be the default columns.
+  param_estimates_compare(
+    .boot_sum$boot_summary, .orig_mod = .orig_mod, probs = probs, na.rm = na.rm
+  )
+}
+
+
+#' @rdname param_estimates_compare
+#' @export
+param_estimates_compare.default <- function(
```

Comment:
Not totally necessary, since this isn't changing current behavior, but it would be nice to do some input checking on `.boot_sum` in this `default` method. Maybe just that it's something that inherits `data.frame` and that it has the columns we expect?

Reply:
I added a data.frame check, but I dont think I can check the columns. It seems like I cant interact with `.compare_cols` if it's not a character vector (I thought it would be an expression, but it just errors out):

In the function:
```r
param_estimates_compare.default <- function(
    .boot_sum,
    .orig_mod = NULL,
    .compare_cols = starts_with(c("THETA", "SIGMA", "OMEGA")),
    probs = c(.5, 0.025, 0.975),
    na.rm = FALSE
){
```

```r
Browse[2]> .compare_cols
Error: `starts_with()` must be used within a *selecting* function.
ℹ See <https://tidyselect.r-lib.org/reference/faq-selection-context.html> for details.

Browse[2]> inherits(.compare_cols, "character")
Error: `starts_with()` must be used within a *selecting* function.
ℹ See <https://tidyselect.r-lib.org/reference/faq-selection-context.html> for details.
In addition: Warning messages:
1: In get(object, envir = currentEnv, inherits = TRUE) :
  restarting interrupted promise evaluation
2: In inherits(.compare_cols, "character") :
  restarting interrupted promise evaluation
```